test/services/ProcessCsvServiceSpec.scala

```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.{HttpResponse, StatusCodes}
+import akka.stream.IOResult
+import akka.stream.alpakka.csv.scaladsl.CsvParsing
+import akka.stream.scaladsl.{FileIO, Sink, Source}
+import akka.testkit.TestKit
+import akka.util.ByteString
+import com.typesafe.config.ConfigFactory
+import controllers.auth.RequestWithOptionalEmpRef
+import helpers.ErsTestHelper
+import models.upscan.{UploadId, UploadedSuccessfully, UpscanCsvFilesCallback, UpscanCsvFilesCallbackList}
+import models.{ERSFileProcessingException, RowValidationResults, SheetErrors}
+import org.mockito.ArgumentMatchers._
+import org.mockito.Mockito._
+import org.scalatest.MustMatchers.convertToAnyMustWrapper
+import org.scalatest.concurrent.{ScalaFutures, TimeLimits}
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.i18n
+import play.api.i18n.{Messages, MessagesImpl}
+import play.api.libs.json.Json
+import play.api.mvc.{AnyContent, DefaultMessagesControllerComponents}
+import play.api.test.FakeRequest
+import uk.gov.hmrc.http.UpstreamErrorResponse
+import uk.gov.hmrc.http.cache.client.CacheMap
+import uk.gov.hmrc.play.test.UnitSpec
+import uk.gov.hmrc.services.validation.DataValidator
+import uk.gov.hmrc.services.validation.models.{Cell, ValidationError}
+import utils.CsvParserUtil
+
+import java.io.File
+import java.nio.file.Files
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+
+class ProcessCsvServiceSpec extends TestKit(ActorSystem("Test")) with UnitSpec with ErsTestHelper with GuiceOneAppPerSuite with TimeLimits with ScalaFutures {
+
+  val fileCopied = new File(System.getProperty("user.dir") + "/test/resources/copy/Other_Grants_V3.csv")
+  if(fileCopied.exists)
+    {
+      fileCopied.getCanonicalFile.delete()
+      fileCopied.getAbsoluteFile.delete()
+    }
```

Comment:
Should all of this be within a function?

Reply:
Good spot, that's actually stuff left over from the old ```CsvFileProcessorSpec``` and is never actually used here so should be deleted